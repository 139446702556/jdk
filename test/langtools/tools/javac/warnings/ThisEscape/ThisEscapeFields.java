/*
 * @test /nodynamiccopyright/
 * @bug 8015831
 * @compile/ref=ThisEscapeFields.out -Xlint:this-escape -XDrawDiagnostics ThisEscapeFields.java
 * @summary Verify 'this' escape detection in field initializers
 */

public class ThisEscapeFields {

    private final int field1 = this.mightLeak1();

    private final int field2 = this.mightLeak2();

    public int mightLeak1() {
        return 123;
    }

    public int mightLeak2() {
        return 456;
    }
}
