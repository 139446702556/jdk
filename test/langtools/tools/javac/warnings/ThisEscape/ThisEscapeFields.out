ThisEscapeFields.java:10:47: compiler.warn.possible.this.escape
ThisEscapeFields.java:12:47: compiler.warn.possible.this.escape
2 warnings
